src/main/java/com/google/jenkins/plugins/storage/ClassicUpload.java

```diff
@@ -34,173 +33,185 @@
-      this(ClassicUpload.class);
+    /**
+     * Construct the classic upload implementation from the base properties and
+     * the glob for matching files.
+     */
+    @DataBoundConstructor
+    public ClassicUpload(String bucket,
+            @Nullable UploadModule module,
+            String pattern,
+            String metadata,
+            // Legacy arguments for backwards compatibility
+            @Deprecated @Nullable String bucketNameWithVars,
+            @Deprecated @Nullable String sourceGlobWithVars) {
+        super(firstNonNull(bucket, bucketNameWithVars), module);
+        this.sourceGlobWithVars = firstNonNull(pattern, sourceGlobWithVars);
+        this.metadata = metadata;
     }
 
-    public DescriptorImpl(
-        Class<? extends ClassicUpload> clazz) {
-      super(clazz);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDetails() {
+        return getPattern();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public String getDisplayName() {
-      return Messages.ClassicUpload_DisplayName();
+    @Nullable
+    protected UploadSpec getInclusions(Run<?, ?> run,
+            FilePath workspace, TaskListener listener) throws UploadException {
+        try {
+
+            String globResolvedVars = StorageUtil
+                    .replaceMacro(getPattern(), run, listener);
+
+        // In order to support absolute globs (e.g. /gagent/metaOutput/std*.txt)
+        // we must identify absolute paths and rebase the "workspace" to be the
+        // root directory and the glob to be relative to root.
+        //
+        // NOTE: FilePath#list complains when the ant-style glob is actually
+        // an absolute path.
+        //
+       // TODO(mattmoor): This will not work on Windows, but short of relying
+        // on Java 7's java.nio.file.Paths there doesn't appear to be a good
+       // platform agnostic approach to this.
+            if (globResolvedVars.startsWith("/")) {
+                workspace = getRoot(workspace);
+            // TODO(mattmoor): Support Windows
+            // Drop the leading "/" in a fashion that should be compatible with
+            // non-Unix platforms (e.g. Windows, which uses drive letters: C:\)
+                globResolvedVars = globResolvedVars.substring(1);
+            }
+
+            FilePath[] inclusions = workspace.list(globResolvedVars);
+            if (inclusions.length == 0) {
+                listener.error(module.prefix(
+                        Messages.ClassicUpload_NoArtifacts(
+                                globResolvedVars)));
+                return null;
+            }
+            listener.getLogger().println(module.prefix(
+                    Messages.ClassicUpload_FoundForPattern(
+                            inclusions.length, getPattern())));
+            return new UploadSpec(workspace, Arrays.asList(inclusions));
+        } catch (InterruptedException e) {
+            throw new UploadException(
```

Comment:
I think your formatter is different from mine.
Could you avoid reformatting the code you are not changing? This would make "blame" more useful.

Reply:
It's last Netbeans formatting and i belive it's better than some old (used in this code).

Reply:
Please undo formatting changes for the lines you are not modifying.
This makes reviews harder and change history less useful.
That is the reason why the [governance document](https://jenkins.io/project/governance/) recommends splitting formatting changes from functional changes.

In this particular case I do not believe formatting changes are warranted at all. We cannot change all the code every time someone comes with different editor settings.